e2e_tests/tests/test_password_reset.py

```diff
@@ -0,0 +1,377 @@
+        self.assertTrue(form.submit_btn.is_enabled())
+
+    def test_error_on_weak_passwords(self):
+        """
+        Test that an error is shown on weak passwords
+        """
+        # Jerry Mouse opens the reset form url
+        self._open_password_reset_with_token(self.reset_password_token.key)
+        self.wait.until(
+            EC.presence_of_element_located((By.TAG_NAME, 'form'))
+        )
+        form = self.form
+        # Jerry enters his new password in both fields and submits
+        form.password_field.send_keys("nibbles")
+        form.password_confirm_field.send_keys("nibbles")
+        form.submit_btn.click()
+        self.wait.until(
+            EC.presence_of_element_located((By.CLASS_NAME, 'alert-danger'))
+        )
+
+        # Jerry sees an alert div with multiple error messages
+        alert_element = form.form.find_element_by_class_name('alert-danger')
+        alert_text = alert_element.text.lower()
+        self.assertTrue(alert_element.is_displayed())
+        self._assert_strong_password_alert(alert_text)
+
+    def test_password_changed_on_strong_password(self):
+        """
+        Tests that the password is correct.
+        Checks that the password is reset by completing login
+        """
+        # Jerry Mouse opens the reset form url
+        self._open_password_reset_with_token(self.reset_password_token.key)
+        self.wait.until(
+            EC.presence_of_element_located((By.TAG_NAME, 'form'))
+        )
+        form = self.form
+        # Jerry enters his new strong password in both fields and submits
+        form.password_field.send_keys(self.strong_password)
+        form.password_confirm_field.send_keys(self.strong_password)
+        form.submit_btn.click()
+        current_url = self.confirm_token_page_url.format(token=self.reset_password_token.key)
+        self.wait.until(
+            EC.url_changes((By.CLASS_NAME, current_url))
+        )
+
+        # Jerry Mouse is redirected to login page with success message.
+        self.assertEqual(self.browser.current_url, self.login_page_url)
+
+        login_form = self.browser.find_element_by_tag_name('form')
+        self.assertTrue(login_form.is_displayed())
+        username_field = login_form.find_element_by_name('username')
+        password_field = login_form.find_element_by_name('password')
+        submit_btn = login_form.find_element_by_tag_name('button')
+        alert_element = login_form.find_element_by_class_name('alert-success')
+
+        self.assertIn('your password has been reset', alert_element.text.lower())
+
+        # Jerry Mouse enters his new credentials and is successfully logged in
+        username_field.send_keys('jerry.mouse')
+        password_field.send_keys(self.strong_password)
+        submit_btn.click()
+        self.wait.until(
+            EC.url_changes((By.CLASS_NAME, self.login_page_url))
+        )
+
+        self.assertEqual(self.browser.current_url, self.theme_preview_url)
```

Comment:
Should we perhaps also do a quick test by trying to log in with the new password?